include/dds/core/xtypes/detail/MemberType.hpp

```diff
@@ -35,25 +35,19 @@ class MemberType
 public:
     MemberType(
             const std::string& name,
-            xtypes::DynamicType& dt)
+            const DynamicType& dynamic_type)
         : name_(name)
-#if(0)
-        , dt_(dt.name(),dt.kind(), dt.annotations())
-#else
-        , dt_(dt)
-#endif
-        , ann_()
+        , dynamic_type_(dynamic_type)
     {}
 
     MemberType(
             const std::string& name,
-            xtypes::DynamicType& dt,
-            xtypes::Annotation& a):
-        name_(name),
-        dt_(dt),
-        ann_()
+            const DynamicType& dynamic_type,
+            Annotation& annotation)
+        : name_(name)
+        , dynamic_type_(dynamic_type)
     {
-        ann_.push_back(a);
+        annotations_.push_back(annotation);
```

Comment:
I thought we said that a class member could have a straight forward shortage of the name.